### DISCUSSION TITLE
DIPY Project 3 'Correlation Tensor Magnetic Resonance Imaging'.

### DISCUSSION THREAD

**Discussion by varsha089** [dipy/dipy#2734] [#GSoC/GSoD]
Hello, @RafaelNH @ShreyasFadnavis

I am Varsha Ahirwar, 3rd year IT undergrad student at IGDTUW, INDIA. I'm interested in contributing to the 'DIPY' as a GSOC 2023 contributor. Specifically, I'm interested in the "Correlation Tensor Magnetic Resonance Imaging" project. I am looking for your guidance to start contributing to this project. I have practical knowledge of python. As this project needed knowledge of 'Diffusion MRI', but I am an engineering student and do not have experience with Diffusion MRI although I am getting familiar with Diffusion MRI through research papers. According to the project description, the experience of diffusion MRI will be a plus as far as I could understand it is not necessary to have, but I am wondering, if will it affect my understanding of the project and performance. 

Thank You.

**Comment by RafaelNH** [comment]
Hi Varsha,

First thanks for reaching us. For this project the essential requirement is to have experience with python programming. Regarding diffusion MRI theory, in this project, the student will be working with 2nd and 4th order tensors to quantify diffusion MRI properties. So as a guidance, before trying to implement the proposed Correlation Tensor Imaging, I will suggest you to give a look to the following simpler and already established diffusion MRI techniques:

1) [Diffusion Tensor Imaging](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1275686/?page=1) that uses a 2nd order tensor to quantify water diffusion anisotropy.

2) [Diffusional Kurtosis Imaging](https://onlinelibrary.wiley.com/doi/10.1002/mrm.20508) that, in addition to the 2nd order tensor, it also uses a 4th order tensor to quantify diffusion non-Gaussian effects.

We already have the implementations of DTI and DKI in DIPY. So another suggestion is to give a look to DTI and DKI codes ([dti.py](https://github.com/dipy/dipy/blob/master/dipy/reconst/dti.py), [dki.py](https://github.com/dipy/dipy/blob/master/dipy/reconst/dki.py)), and DTI and DKI tutorials ([example-reconst-dti](https://dipy.org/documentation/1.6.0./examples_built/reconst_dti/#example-reconst-dti), [example-reconst-dki](https://dipy.org/documentation/1.6.0./examples_built/reconst_dki/#example-reconst-dki)). At the end of the GSoC project it is expected that the student will implement the equivalent code and tutorial for CTI, so see from the codes of previous implemented techniques if this is something you will be excited to be working on.

**Comment by varsha089** [comment]
Hello @RafaelNH 

Thank you for your response and valuable guidance. I appreciate your suggestions and will definitely look into the simpler and already established diffusion MRI techniques like DTI and DKI. I will also explore the implementations of DTI and DKI in DIPY along with their tutorials.

I will work hard to gain a good understanding of Diffusion MRI theory and the implementation of the proposed Correlation Tensor Imaging. I am excited to contribute to the DIPY project and I believe that with your guidance, I will be able to perform well.

Thank you once again for your time and guidance.

Best regards,
Varsha Ahirwar

**Comment by varsha089** [comment]
Hello, @RafaelNH 
As you mentioned that the essential requirement of this project is to have experience with python programming. So while learning about the implementations of DTI and DKI in DIPY and 2nd and 4th-order tensors to quantify diffusion MRI properties.
Are there any milestones that contributors are expected to achieve for submitting the proposal of this project as a potential GSOC 2023 contributor?

**Comment by skoudoro** [comment]
Hi @varsha089,

> Are there any milestones that contributors are expected to achieve for submitting the proposal of this project as a potential GSOC 2023 contributor?


Please, follow our wiki page guideline and Python Software Foundation (psf) guideline concerning the milestone. See below some citations of the guideline:

-  for DIPY wiki :  https://github.com/dipy/dipy/wiki/Google-Summer-of-Code-2023#advice 

> Potential candidates should take a look at the guidelines on how to [contribute to DIPY](https://github.com/dipy/dipy/blob/master/CONTRIBUTING.md). Making a small enhancement/bugfix/documentation fix/etc to DIPY already before applying for the GSoC can help you get some idea how things would work during the GSoC. The fix does not need to be related to your proposal. We have and will continue adding some beginner-friendly issues in Github. You can see some of them [here](https://github.com/dipy/dipy/issues?q=is%3Aissue+is%3Aopen+label%3ABeginner-friendly) or all the issues [here](https://github.com/dipy/dipy/issues).

- PSF guideline: https://python-gsoc.org/contributors.html
